GOVERNANCE.md

```diff
@@ -107,6 +107,28 @@ Lazy consensus does _not_ apply to the process of:
 
 * Removal of maintainers from Velero
 
+## Deprecation Policy
+
+### Deprecation Process
+
+Any contributor may introduce a request to deprecate a feature or an option of a feature by opening a feature request issue in the vmware-tanzu/velero GitHub project. The issue should describe why the feature is no longer needed or has become detrimental to Velero, as well as whether and how it has been superseded. The submitter should give as much detail as possible.
```

Comment:
For Velero, on one hand, it has a feature that may be represented by code, modules or interfaces; on the other hand, it has the permanent data that created by the feature.
If the feature means all the code/modules/interfaces, once the feature is removed, the data created by it will not be accessible.
However, as a backup/restore tool, the backed up data are usually demanded for long-time preservation, i.e., years or decades.

Therefore, for a feature that generates backup data with a unique format, for example:

- A uploader or backup repository like Restic
- A encryption/compression algorithm for backup

along with the deprecation of a feature, should we clarify the end-of-support policy for the backup data?

Reply:
@Lyndon-Li That's a good point. It may be that for something like restic, as one example, we need to keep restore support longer than backup support. For example, if we deprecate restic in 1.13, we could remove backup support in 1.15, but we may need to keep restore support until 1.17. In other words, Velero 1.14 would still allow new backups to use restic, and restores with restic, but in Velero 1.15 or 1.16, new backups will not be able to use restic, but a velero 1.12 or 1.13 backup with restic data could still be restored.

Or in more general terms, if a deprecated feature must remain available for general use for two releases after deprecation, perhaps a backup that uses this deprecated feature must be fully-supported for restore for four releases.

Reply:
@weshayutin 
Could you help to include this part in the PR?
For the details, personal I agree with @sseago's proposal above.

cc @reasonerjt @yanji09

Reply:
@Lyndon-Li Done, added a note regarding this.

Reply:
Thank you! @shubham-pampattiwar